mapie/conformity_scores/conformity_scores.py

```diff
@@ -158,20 +183,151 @@ def get_conformity_scores(
+            Values from which the quantile is computed, it can be the
+            conformity scores or the conformity scores aggregated with
+            the predictions.
+
+        alpha_np: NDArray
+            NDArray of floats between ``0`` and ``1``, represents the
+            uncertainty of the confidence interval.
+
+        axis: int
+            The axis from which to compute the quantile.
+
+        method: str
+            ``"higher"`` or ``"lower"`` the method to compute the quantile.
+
+        Returns
+        -------
+        NDArray
+            Lower and upper quantile of the prediction intervals.
+            These quantiles are identical if the score is not symmetrical.
+        """
+        quantile = np.column_stack([
+            np_nanquantile(
+                values.astype(float),
+                _alpha,
+                axis=axis,
+                method=method
+            )
+            for _alpha in alpha_np
+        ])
+        return quantile
+
+    def get_bounds(
+        self,
+        X: ArrayLike,
+        estimator: RegressorMixin,
+        conformity_scores: NDArray,
+        alpha_np: NDArray,
+        ensemble: bool,
+        method: str
+    ) -> Tuple[NDArray, NDArray, NDArray]:
+        """
+        Compute bounds of the prediction intervals from the observed values,
+        the estimator of MapieRegressor and the conformity scores.
+
+        Parameters
+        ----------
+        X: ArrayLike
+            Observed feature values.
+
+        estimator: RegressorMixin
+            Estimator that is fitted to predict y from X.
+
+        conformity_scores: ArrayLike
+            Conformity scores.
+
+        alpha_np: NDArray
+            NDArray of floats between ``0`` and ``1``, represents the
+            uncertainty of the confidence interval.
+
+        ensemble: bool
+            Boolean determining whether the predictions are ensembled or not.
+
+        method: str
+            The method parameter of MapieRegressor.
+
+        Returns
+        -------
+        Tuple[NDArray, NDArray, NDArray]
+            - The predictions itself. (y_pred)
+            - The lower bounds of the prediction intervals.
+            - The upper bounds of the prediction intervals.
+        """
+        y_pred, y_pred_low, y_pred_up = estimator.predict(X, ensemble)
+
+        signed = -1 if self.sym else 1
+        alpha_low = alpha_np if self.sym else alpha_np / 2
+        alpha_up = 1 - alpha_np if self.sym else 1 - alpha_np / 2
+
+        if method == "plus":
+            bound_low = self._get_quantile(
+                self.get_estimation_distribution(
+                    X, y_pred_low, signed * conformity_scores
+                ),
+                alpha_low,
+                axis=1,
+                method="lower"
+            )
+            bound_up = self._get_quantile(
+                self.get_estimation_distribution(
+                    X, y_pred_up, conformity_scores
+                ),
```

Comment:
UPD: calculate it before calculating the quantile (break down the complete calculation into two steps as in the else condition).